cmd/harvest/harvest.go

```diff
@@ -319,12 +327,61 @@ func checkPollerIdentity(cmdline, pollerName string) bool {
 		return false
 	}
 
-	if y := strings.Fields(x[1]); len(y) == 0 || y[0] != pollerName {
-		return false
+	if verifyPollerName {
+		if y := strings.Fields(x[1]); len(y) == 0 || y[0] != pollerName {
+			return false
+		}
 	}
 	return true
 }
 
+func stopGhostPollers(search string, skipPoller []string) {
+	result, err := util.GetProcessPid(search)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	pids := util.RemoveEmptyStrings(strings.Split(result, "\n"))
+	for _, pid := range pids {
+		p, err := strconv.Atoi(strings.TrimSpace(pid))
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+		c, err := util.GetCmdLine(p)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		if checkPollerIdentity(c, "", false) {
+			// skip if this poller is defined in harvest config
+			var skip bool
+			for _, s := range skipPoller {
+				if strings.Contains(c, s) {
+					skip = true
+					break
+				}
+			}
+			// if poller doesn't exists in harvest config
+			if !skip {
+				proc, err := os.FindProcess(p)
+				if err != nil {
+					fmt.Println(err)
```

Comment:
more context about not finding pid we expect?

Reply:
done